controllers/storage.go

```diff
@@ -355,8 +355,10 @@ func (s *Storage) CopyFromPath(artifact *sourcev1.Artifact, path string) (err er
 	if err != nil {
 		return err
 	}
-	defer f.Close()
-	return s.Copy(artifact, f)
+	if err = s.Copy(artifact, f); err != nil {
+		return err
+	}
+	return f.Close()
```

Comment:
If the Copy func fails, `f.Close` won't be called, potentially leaving `f` open until garbage collected.
You probably want to defer it whilst handling the error before the call to copy:

```
    defer func() {
        if cerr := f.Close(); cerr != nil {
            err = cerr
        }
    }()
```

Reply:
Ah thanks for the suggestion!